File: C#.Net/JDI/Web/JDI Web/Selenium/Elements/Complex/Table/RowColumn.cs
﻿using System;

namespace JDI_Web.Selenium.Elements.Complex.Table
{
    public class RowColumn
    {
        public int Num { get; set; }
        public string Name { get; set; }

        public RowColumn(int num, string name)
        {
            Num = num;
            Name = name;
        }

        public bool HasName => !string.IsNullOrEmpty(Name);

        public T Get<T>(Func<RowColumn, T> action)
        {
            return action.Invoke(this);
        }
        public T Get<T>(Func<string, T> nameAction, Func<int, T> numAction)
        {
            return HasName ? nameAction.Invoke(Name) : numAction.Invoke(Num);
        }
    }
}

File: Python/JDI/web/selenium/elements/complex/table/row_column.py
class RowColumn:
    num = None
    name = None

    def __init__(self, val):
        if isinstance(val, str):
            self.name = val
        elif isinstance(val, int):
            self.num = val

    def get(self, name_action, num_action):
        return name_action(self.name) if self.has_name() else num_action(self.num)

    def has_name(self):
        return self.name is not None and self.name != ""
